Common/src/geometry/meshreader/CSU2ASCIIMeshReaderFVM.cpp

```diff
@@ -122,128 +129,140 @@ void CSU2ASCIIMeshReaderFVM::ReadMetadata() {
 
-      su2double AoS_Current = config->GetAoS() + AoS_Offset;
-
-      if (config->GetDiscard_InFiles() == false) {
-        if ((rank == MASTER_NODE) && (AoS_Offset != 0.0))  {
-          cout.precision(6);
-          cout << fixed <<"WARNING: AoS in the config file (" << config->GetAoS() << " deg.) +" << endl;
+      if (AoS_Offset != 0.0) {
+        if (!config->GetDiscard_InFiles()) {
+          cout << "WARNING: AoS in the config file (" << config->GetAoS() << " deg.) +\n";
           cout << "         AoS offset in mesh file (" << AoS_Offset << " deg.) = " << AoS_Current << " deg." << endl;
         }
-        config->SetAoS_Offset(AoS_Offset);
-        config->SetAoS(AoS_Current);
-      }
-      else {
-        if ((rank == MASTER_NODE) && (AoS_Offset != 0.0))
-          cout <<"WARNING: Discarding the AoS offset in the geometry file." << endl;
+        else {
+          cout << "WARNING: Discarding the AoS offset in the mesh file." << endl;
+        }
       }
-
+      continue;
     }
 
-    position = text_line.find ("NPOIN=",0);
-    if (position != string::npos) {
+    if (!foundNPOIN && text_line.find ("NPOIN=",0) != string::npos) {
       text_line.erase (0,6);
       numberOfGlobalPoints = atoi(text_line.c_str());
-      for (unsigned long iPoint = 0; iPoint < numberOfGlobalPoints; iPoint++)
-        getline (mesh_file, text_line);
+
+      /* If the points were found first, read them, otherwise just consume the lines. */
+      if (single_pass && foundNDIME && current_section_idx == 0) {
+        single_pass_active = true;
+        ReadPointCoordinates(true);
+      }
+      else {
+        for (auto iPoint = 0ul; iPoint < numberOfGlobalPoints; iPoint++)
+          getline (mesh_file, text_line);
+      }
+      SectionOrder[current_section_idx++] = FileSection::POINTS;
       foundNPOIN = true;
+      continue;
     }
 
-    position = text_line.find ("NELEM=",0);
-    if (position != string::npos) {
+    if (!foundNELEM && text_line.find ("NELEM=",0) != string::npos) {
       text_line.erase (0,6);
       numberOfGlobalElements = atoi(text_line.c_str());
-      for (unsigned long iElem = 0; iElem < numberOfGlobalElements; iElem++)
-        getline (mesh_file, text_line);
+
+      if (single_pass_active) {
+        ReadVolumeElementConnectivity(true);
+      }
+      else {
+        for (auto iElem = 0ul; iElem < numberOfGlobalElements; iElem++)
+          getline (mesh_file, text_line);
+      }
+      SectionOrder[current_section_idx++] = FileSection::ELEMENTS;
       foundNELEM = true;
+      continue;
     }
 
-    position = text_line.find ("NMARK=",0);
-    if (position != string::npos) {
+    if (!foundNMARK && text_line.find ("NMARK=",0) != string::npos) {
       text_line.erase (0,6);
       numberOfMarkers = atoi(text_line.c_str());
+
+      if (current_section_idx != 2) {
+        SU2_MPI::Error("Markers must be listed after points and elements in the SU2 mesh file.", CURRENT_FUNCTION);
+      }
```

Comment:
This PR introduces this restriction. It could be avoided if we stop checking for v6-style periodicity in the mesh file.